File: frontend/src/shared/types/note.ts
// Domain types for note operations — backend-agnostic

export interface CreateNoteInput {
  title: string;
  content?: string;
  tags?: string[];
  visibility?: 'private' | 'public';
}

export interface UpdateNoteInput {
  title?: string;
  content?: string;
  tags?: string[];
  visibility?: 'private' | 'public';
}

File: backend/SmartLearn.Domain/Entities/Note.cs
using SmartLearn.Domain.Common;
using SmartLearn.Domain.Enums;

namespace SmartLearn.Domain.Entities;

public class Note : BaseEntity
{
    public Guid OwnerId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Content { get; set; } // BlockNote JSON
    public string Tags { get; set; } = "[]"; // JSON array of strings
    public NoteVisibility Visibility { get; set; } = NoteVisibility.Private;

    // ── Navigation ──────────────────────────────────────────
    public User Owner { get; set; } = null!;
}
